examples/gno.land/r/demo/gnolotto_factory/gnolotto_factory.gno

```diff
@@ -0,0 +1,239 @@
+package gnolotto_factory
+
+import (
+	"bytes"
+	"std"
+	"strconv"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/gnolotto"
+	"gno.land/p/demo/ufmt"
+)
+
+var lotteries *avl.Tree
+
+// Replace this address with your address
+var admin std.Address = "g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5"
+
+// Initializes the lottery AVL tree
+func init() {
+	lotteries = avl.NewTree()
+}
+
+// Creates a new lottery, only callable by admin.
+func CreateLottery(drawTime int64, prizePool int64) (int, string) {
+	sentCoins := std.GetOrigSend()
+	amount := sentCoins.AmountOf("ugnot")
+	banker := std.GetBanker(std.BankerTypeRealmSend)
+	send := std.Coins{{"ugnot", int64(amount)}}
```

Comment:
Casting to `int64` is redundant as coins.AmountOf returns `int64`

Reply:
✅